pkg/skaffold/build/kaniko/sources/sources.go

```diff
@@ -70,6 +70,10 @@ func podTemplate(cfg *latest.KanikoBuild, args []string) *v1.Pod {
 							Name:      constants.DefaultKanikoSecretName,
 							MountPath: "/secret",
 						},
```

Comment:
So why not overload the credentials and mount them as the normal creds in `/secret` as well as in `/root/.docker/config.json`?

In my case this allowed me to set the `kanikoSecret: ~/.docker/config.json`  inside of the kaniko builder configuration.

Reply:
Interesting idea. My thoughts: 
1.) However there is no way _both_ can work. What if you want to build two artifacts, one with a GCR repo, the other one using Docker? The first would want to use the `GOOGLE_APPLICATION_CREDENTIALS` env var, the latter would use the config.json. 
2.) based on https://github.com/GoogleContainerTools/kaniko#pushing-to-amazon-ecr I see that it should be /kaniko/.docker/config.json and could be a configmap... @priyawadhwa can you advise? 
3.) What if we introduce a `mountDockerConfig` boolean flag, by default `false` and a `dockerConfigPath:` string  flag pointing to `~/.docker/config.json` by default instead?

Reply:
Yup option 3 SGTM -- basically skaffold.yaml should have separate options for registry specific credentials and the dockerconfig in case both are needed, as @balopat said. We could create a configmap or another secret (probably better) for the dockerconfig and mount it into `/kaniko/.docker/config.json` (that's where kaniko [expects](https://github.com/GoogleContainerTools/kaniko/blob/master/deploy/Dockerfile#L38) to find it)

Reply:
Sorry for the delay. I wasn't intending for the configuration to be reused, I only did it to simplify the work required to test the concept. Realistically option 3 is probably be the best.